pkg/provenance/sign_test.go

```diff
@@ -266,6 +270,85 @@ func TestClearSign(t *testing.T) {
 	}
 }
 
+func TestMixedKeyringRSASigningAndVerification(t *testing.T) {
+	signer, err := NewFromFiles(testKeyfile, testMixedKeyring)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(signer.KeyRing) == 0 {
+		t.Fatal("expected signer keyring to be loaded")
+	}
+
+	hasEdDSA := false
+	for _, entity := range signer.KeyRing {
+		if entity.PrimaryKey != nil && entity.PrimaryKey.PubKeyAlgo == packet.PubKeyAlgoEdDSA {
+			hasEdDSA = true
+			break
+		}
+
+		for _, subkey := range entity.Subkeys {
+			if subkey.PublicKey != nil && subkey.PublicKey.PubKeyAlgo == packet.PubKeyAlgoEdDSA {
+				hasEdDSA = true
+				break
+			}
+		}
+
+		if hasEdDSA {
+			break
+		}
+	}
+
+	if !hasEdDSA {
+		t.Fatalf("expected %s to include an Ed25519 public key", testMixedKeyring)
+	}
```

Comment:
testify/assert would clean these up:

```
assert.True(t, hasEdDSA)
```

and e.g. below
```
assert.NotNil(t, signer.Entity)
```

(messages become optional IMHO, as the single line assert typically conveys enough info)

Reply:
I created https://github.com/helm/helm/pull/31326 as a follow up to address this